htlcswitch/switch.go

```diff
@@ -1851,7 +1853,7 @@ func (s *Switch) getLinkByShortID(chanID lnwire.ShortChannelID) (ChannelLink, er
 
 // RemoveLink is used to initiate the handling of the remove link command. The
 // request will be propagated/handled to/in the main goroutine.
-func (s *Switch) RemoveLink(chanID lnwire.ChannelID) error {
+func (s *Switch) RemoveLink(chanID lnwire.ChannelID) (ChannelLink, error) {
```

Comment:
`godoc` should be updated to reflect that it is now the caller's responsibility to stop the link.